content/blog/2025/oddcontrast-guide.md

```diff
@@ -0,0 +1,184 @@
+    img: 'blog/2025/oddcontrast-swaps.png',
+    alt: 'use the arrow icon to swap foreground and background colors'
+  }],
+  class='extend-large'
+) }}
+
+Once you are happy with your colors, you can easily copy your preferred color
+notation to the clipboard with a single click.
+
+{{ embed.figure(
+  data=[{
+    img: 'blog/2025/oddcontrast-clipboard.png',
+    alt: 'the copy to clipboard flow'
+  }]
+) }}
+
+## Visualize the in-gamut range
+
+OddContrast allows you to choose whether or not to display a color gamut for
+visual reference, and if so, which specific gamut you'd like to see. Once a
+gamut is selected, the color sliders will display the "in-gamut" boundaries and
+will update with the color values. To preview this, select the sRGB option in
+the "Show Gamut" field, edit the color sliders and see how every shift in color
+will affect the other channels.
+
+{{ embed.figure(
+  data=[{
+    img: 'blog/2025/oddcontrast-srgb-gamut.jpg',
+    alt: 'use the arrow icon to swap foreground and background colors'
+  }]
+) }}
+
+<div class="contain">
+{{ embed.figure(
+  data=[{
+    img: 'blog/2025/oddcontrast-gamut-warning.jpg',
+    alt: 'use the arrow icon to swap foreground and background colors'
+  }],
+    class='align-left'
+) }}
+
+If your color is outside the selected gamut,
+a warning will be displayed over the swatch.
+A separate warning will appear near the color notations
+for each affected color space
+regardless of your gamut display setting.
+</div>
+
+## Saving and sharing your color combinations
+
+OddContrast automatically updates the URL as you make changes to the colors.
+This dynamic URL allows you to easily share color combinations with others or
+bookmark them for your own future use.
+
+## Contrast ratios and transparency
+
+Using the [WCAG 2 contrast ratio requirements](https://webaim.org/articles/contrast/#ratio),
+OddContrast displays a pass or fail message for meeting the minimum (Level AA)
+and enhanced (Level AAA) values for normal and large text.
+A warning is shown if the the colors include an alpha value lower than one
+as WCAG 2 does not account for transparency.
+
+OddContrast approximates a ratio by premultiplying a semi-transparent
+foreground color in the sRGB space. In practice,
+the displayed foreground color may vary depending on the display and browser.
+
+Without knowing what is behind a background color with transparency,
+OddContrast cannot estimate the contrast ratio.
+If the background color is not opaque,
+the contrast ratio is computed without background or foreground opacity.
+
+{{ embed.figure(
+  data=[{
+    img: 'blog/2025/oddcontrast-alpha.jpg',
+    alt: 'a warning is shown when using the alpha channel'
+  }]
+) }}
+
+## For fun
+
+To add some whimsy, OddContrast proudly showcases your chosen color
+combination within its very own logo.
+
+{{ embed.figure(
+  data=[{
+    img: 'blog/2025/oddcontrast-inline-logos.png',
+    alt: 'OddContrast logo with different colors'
+  }]
+) }}
+
```

Comment:
I need to give these a background as they blend in with the dark mode background.

Reply:
![Screenshot 2025-06-05 at 12 09 09 PM](https://github.com/user-attachments/assets/dfb572e2-2564-438e-bd0b-0ea792bd136d)
 
updated